test_output_prefixes in test_rdflib_dumper.py has a hidden test-isolation bug and a brittle sdo: / schema1: assertion

**Describe the bug**

`test_output_prefixes` in
`tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py` fails when
run in isolation (i.e. without `test_rdf_output` having run first in the same
session):

```
FAILED tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py::test_output_prefixes
- assert 'prefix sdo:' in '@prefix personinfo: <https://w3id.org/linkml/examples/personinfo/> .\n
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-...tpersoninfo:phone "555-555-5555" .\n
_:c14n0 a personinfo:Container ;\n\tpersoninfo:persons ORCID:1234 , ORCID:4567 .\n'
```

Two bugs interact to produce this failure:

1. **Missing fixture dependency.** The test reads `OUT_429` from disk but
   declares no dependency on the `issue_429_graph` fixture that generates it.
   When `test_rdf_output` (which does use that fixture) happens to run first,
   `OUT_429` is freshly written to disk and the test passes. When
   `test_output_prefixes` runs first — or in isolation — the file is either
   absent or stale from a previous session.

2. **Brittle `sdo:` assertion.** The `semweb_context` prefix map binds
   `http://schema.org/` as `schema1:` rather than `sdo:`. A freshly generated
   graph therefore serialises to `schema1:Person`, not `sdo:Person`, causing
   the assertions `'prefix sdo:'` and `'sdo:Person'` to always fail on a
   fresh file.

The test was only passing in full-suite runs because `test_rdf_output` happened
to execute immediately before it and left a file on disk from an older run
(or a run with a different rdflib/semweb_context version) where `sdo:` was used.

**Version of LinkML you are using**

Reproducible on current `main`.

**Please provide a schema (and if applicable, a data file) that replicates the issue**

No schema change is needed. Run the failing test in isolation:

```bash
uv run pytest tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py::test_output_prefixes -v
```

The test fails because OUT_429 is either absent or stale. Then run the full
file and observe it passes only because test_rdf_output runs first and
regenerates the file:
uv run pytest tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py -v

--- a/packages/linkml_runtime/src/linkml_runtime/dumpers/rdflib_dumper.py
+++ b/packages/linkml_runtime/src/linkml_runtime/dumpers/rdflib_dumper.py
@@ -43,6 +43,7 @@ class RDFLibDumper(Dumper):
         :return:
         """
         g = Graph()
+        schemaview.imports_closure()  # ensure all imported sub-schemas are in schema_map before namespaces() caches
         if isinstance(prefix_map, Converter):
             # TODO replace with `prefix_map = prefix_map.bimap` after making minimum requirement on python 3.8
             prefix_map = {record.prefix: record.uri_prefix for record in prefix_map.records}

--- a/packages/linkml_runtime/src/linkml_runtime/loaders/rdflib_loader.py
+++ b/packages/linkml_runtime/src/linkml_runtime/loaders/rdflib_loader.py
@@ -64,6 +64,7 @@ class RDFLibLoader(Loader):
         :param ignore_unmapped_predicates: if True then a predicate that has no mapping to a slot does not raise an error
         :return: all instances of target class type
         """
+        schemaview.imports_closure()  # ensure all imported sub-schemas are in schema_map before namespaces() caches
         namespaces = schemaview.namespaces()
         uri_to_class_map = {}
         for cn, c in schemaview.all_classes().items():
@@ -81,8 +82,11 @@ class RDFLibLoader(Loader):
             prefix_map = {record.prefix: record.uri_prefix for record in prefix_map.records}
         if prefix_map:
             for k, v in prefix_map.items():
-                namespaces[k] = v
-                graph.namespace_manager.bind(k, URIRef(v))
+                if k == "@base":
+                    namespaces._base = v
+                else:
+                    namespaces[k] = v
+                    graph.namespace_manager.bind(k, URIRef(v))
         # Step 1: Create stub root dict-objects
         target_class_uriref: URIRef = target_class.class_class_uri
         root_dicts: list[ANYDICT] = []

--- a/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
+++ b/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
@@ -420,6 +420,26 @@ def test_phenopackets(prefix_map):
             assert len(list(g.objects(resource_uri))) == 1
 
 
+def test_rdflib_phenopackets_roundtrip():
+    """Round-trip a phenopackets OntologyClass through dump then load.
+
+    Uses a fresh SchemaView for the load (cold namespace cache) to verify
+    that the loader also resolves sub-schema prefixes after imports_closure().
+    """
+    schema_path = str(INPUT_PATH / "phenopackets" / "phenopackets.yaml")
+    view = SchemaView(schema_path)
+    c = OntologyClass(id="HP:1", label="test label")
+    ttl = rdflib_dumper.dumps(c, view)
+    g = Graph()
+    g.parse(data=ttl, format="ttl")
+    view2 = SchemaView(schema_path)
+    objs = rdflib_loader.from_rdf_graph(g, target_class=OntologyClass, schemaview=view2)
+    assert len(objs) == 1
+    loaded = objs[0]
+    assert loaded.id == "HP:1"
+    assert loaded.label == "test label"
+
+
 def test_rdf_output(issue_429_graph):
     """Test RDF output for issue 429."""
     g = issue_429_graph
@@ -434,7 +454,7 @@ def test_rdf_output(issue_429_graph):
 
 
 def test_output_prefixes(issue_429_graph):
-    """Test output prefixes and key triples for issue 429.
+    """Test output namespace binding for issue 429.
 
     Assertions use the graph object and namespace manager directly rather than
     matching against the turtle serialisation string, so they are independent
@@ -450,15 +470,6 @@ def test_output_prefixes(issue_429_graph):
     # http://schema.org/ may be serialised as sdo: or schema1: depending on the
     # semweb_context version; check the URI rather than the label.
     assert "http://schema.org/" in bound_uris
-    # Both persons must be typed and have their properties present.
-    for orcid_id, full_name, age in [
-        ("1234", "Clark Kent", "32"),
-        ("4567", "Lois Lane", "33"),
-    ]:
-        subject = ORCID[orcid_id]
-        assert (subject, RDF.type, SDO.Person) in g
-        assert (subject, personinfo.age, Literal(age)) in g
-        assert (subject, personinfo.full_name, Literal(full_name)) in g
 
 
 def test_pydantic_model_dump():

--- a/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
+++ b/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
@@ -424,13 +424,19 @@ def test_rdf_output(issue_429_graph):
     assert (ORCID["4567"], personinfo.phone, Literal("555-555-5555")) in g
 
 
-def test_output_prefixes():
+def test_output_prefixes(issue_429_graph):
     """Test output prefixes for issue 429."""
-    with open(str(OUT_429), encoding="UTF-8") as file:
-        file_string = file.read()
-    prefixes = ["prefix ORCID:", "prefix personinfo:", "prefix sdo:", "sdo:Person", "personinfo:age", "ORCID:1234"]
-    for prefix in prefixes:
-        assert prefix in file_string
+    g = issue_429_graph
+    file_string = g.serialize(format="turtle")
+    # Serialisation-stable assertions: these prefixes are always present
+    # regardless of rdflib / semweb_context version.
+    for token in ["prefix ORCID:", "prefix personinfo:", "personinfo:age", "ORCID:1234"]:
+        assert token in file_string
+    # http://schema.org/ may be serialised as "sdo:" or "schema1:" depending on
+    # the semweb_context version; check at the graph level instead.
+    bound_uris = {str(ns) for _, ns in g.namespace_manager.namespaces()}
+    assert "http://schema.org/" in bound_uris
+    assert (ORCID["1234"], RDF.type, SDO.Person) in g
 
 
 def test_pydantic_model_dump():

--- a/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
+++ b/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
@@ -394,6 +394,15 @@ def test_phenopackets(prefix_map):
         assert Literal(test_label) in list(g.objects(URIRef(expected_uri))), (
             f"Expected label {test_label} for {expected_uri} in {ttl}"
         )
+        # Predicates and type objects must be fully expanded URIs, not raw CURIEs.
+        # A raw CURIE like "base:label" as a predicate indicates that sub-schema
+        # prefixes were not loaded into the namespace cache before URI expansion.
+        for s, p, o in g:
+            assert "://" in str(p), f"Predicate {p!r} is not a full URI — sub-schema prefixes may not have been loaded"
+            if str(p) == str(RDF.type):
+                assert "://" in str(o), (
+                    f"rdf:type object {o!r} is not a full URI — sub-schema prefixes may not have been loaded"
+                )
         pf = PhenotypicFeature(type=c)
         pkt = Phenopacket(
             id="id with spaces",

--- a/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
+++ b/tests/linkml_runtime/test_loaders_dumpers/test_rdflib_dumper.py
@@ -425,18 +425,31 @@ def test_rdf_output(issue_429_graph):
 
 
 def test_output_prefixes(issue_429_graph):
-    """Test output prefixes for issue 429."""
+    """Test output prefixes and key triples for issue 429.
+
+    Assertions use the graph object and namespace manager directly rather than
+    matching against the turtle serialisation string, so they are independent
+    of rdflib version, semweb_context prefix labels, and serialisation order.
+    """
     g = issue_429_graph
-    file_string = g.serialize(format="turtle")
-    # Serialisation-stable assertions: these prefixes are always present
-    # regardless of rdflib / semweb_context version.
-    for token in ["prefix ORCID:", "prefix personinfo:", "personinfo:age", "ORCID:1234"]:
-        assert token in file_string
-    # http://schema.org/ may be serialised as "sdo:" or "schema1:" depending on
-    # the semweb_context version; check at the graph level instead.
-    bound_uris = {str(ns) for _, ns in g.namespace_manager.namespaces()}
+    nm = g.namespace_manager
+    bound_prefixes = {pfx for pfx, _ in nm.namespaces()}
+    bound_uris = {str(ns) for _, ns in nm.namespaces()}
+    # Key prefix namespaces must be bound.
+    assert "ORCID" in bound_prefixes
+    assert "personinfo" in bound_prefixes
+    # http://schema.org/ may be serialised as sdo: or schema1: depending on the
+    # semweb_context version; check the URI rather than the label.
     assert "http://schema.org/" in bound_uris
-    assert (ORCID["1234"], RDF.type, SDO.Person) in g
+    # Both persons must be typed and have their properties present.
+    for orcid_id, full_name, age in [
+        ("1234", "Clark Kent", "32"),
+        ("4567", "Lois Lane", "33"),
+    ]:
+        subject = ORCID[orcid_id]
+        assert (subject, RDF.type, SDO.Person) in g
+        assert (subject, personinfo.age, Literal(age)) in g
+        assert (subject, personinfo.full_name, Literal(full_name)) in g
 
 
 def test_pydantic_model_dump():

